apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnType.tsx

```diff
@@ -155,104 +172,125 @@ const ColumnType = ({
-            <Listbox.Option
-              key={enumType.name}
-              value={enumType.name}
-              label={enumType.name}
-              addOnBefore={() => {
-                return <ListPlus size={16} className="text-foreground" strokeWidth={1.5} />
-              }}
-            >
-              <div className="flex items-center space-x-4">
-                <p className="text-foreground">{enumType.name}</p>
-                {enumType.comment !== undefined && (
-                  <p className="text-foreground-lighter">{enumType.comment}</p>
-                )}
+      <Popover_Shadcn_ open={open} onOpenChange={setOpen}>
+        <PopoverTrigger_Shadcn_ asChild>
+          <Button
+            type="default"
+            role="combobox"
+            size={'small'}
+            aria-expanded={open}
+            className="w-full justify-between"
+            iconRight={<ChevronsUpDown className="ml-2 h-4 w-4 shrink-0 opacity-50" />}
+          >
+            {value ? (
+              <div className="flex gap-2">
+                <span>{inferIcon(availableTypes.find((type) => type === value) ?? '')}</span>
+                {availableTypes.find((type) => type === value)}
               </div>
-            </Listbox.Option>
-          ))
-        ) : (
-          <></>
-        )}
+            ) : (
+              'Choose a column type...'
+            )}
+          </Button>
+        </PopoverTrigger_Shadcn_>
+        <PopoverContent_Shadcn_ className="w-[460px] p-0" side="bottom" align="center">
+          <ScrollArea className="h-[335px]">
+            <Command_Shadcn_>
+              <CommandInput_Shadcn_ placeholder="Search types..." />
+              <CommandEmpty_Shadcn_>Type not found.</CommandEmpty_Shadcn_>
 
-        <Listbox.Option disabled value="header-2" label="header-2">
-          PostgreSQL Data Types
-        </Listbox.Option>
+              <CommandList_Shadcn_>
+                <CommandGroup_Shadcn_>
+                  {POSTGRES_DATA_TYPE_OPTIONS.map((option: PostgresDataTypeOption) => (
+                    <CommandItem_Shadcn_
+                      key={option.name}
+                      value={option.name}
+                      className={cn('relative', option.name === value ? 'bg-surface-200' : '')}
+                      onSelect={(value: string) => {
+                        onOptionSelect(value)
+                        setOpen(false)
+                      }}
+                    >
+                      <div className="flex items-center gap-2 pr-6">
+                        <span>{inferIcon(option.type)}</span>
+                        <span className="text-foreground">{option.name}</span>
+                        <span className="text-foreground-lighter">{option.description}</span>
+                      </div>
+                      <span className="absolute right-3 top-2">
+                        {option.name === value ? (
+                          <Check className="text-brand-500" size={14} />
```

Comment:
please use foreground, brand color only for CTAs

Reply:
I'll get something better setup, a reusable check to use for this and the Select component